jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/BaseBidirectionalShortestPathAlgorithm.java

```diff
@@ -100,20 +100,23 @@ public BaseBidirectionalShortestPathAlgorithm(Graph<V, E> graph)
 
     /**
      * Base class of the search frontier used by bidirectional shortest path algorithms.
+     *
+     * @param <V1> vertices type
+     * @param <E1> edges type
      */
-    abstract class BaseSearchFrontier
+    abstract static class BaseSearchFrontier<V1, E1>
```

Comment:
Why use V1,E1 and not V,E?  When the class is static the generic parameters are not dependent on the out class generic parameters, so they can be called with the same name.

Reply:
Good point. The same for `DijkstraSearchFrontier` in `BidirectionalDijkstraShortestPath`.